decision_service/include/ds_driver.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "ds_ranking_response.h"
+#include "ds_config_collection.h"
+#include "ds_api_status.h"
+#include <memory>
+
+namespace decision_service {
+
+	class driver_impl;
+
+	//decision service client
+	class driver {
+
+	public:
+    using error_fn = void(*)(const api_status&, void*);
+    explicit driver(const utility::config_collection& config, error_fn fn = nullptr, void* err_context = nullptr);
+		~driver();
+
+		// request the decision service, in order to rank the N actions provided in the context_json string
+		int ranking_request(const char * uuid, const char * context_json, ranking_response&, api_status* = nullptr);
```

Comment:
>ranking_request [](start = 6, length = 15)

rename to choose_rank

Reply:
Done

---
In reply to: [187626574](https://github.com/JohnLangford/vowpal_wabbit/pull/1479#discussion_r187626574) [](ancestors = 187626574)